rules/contrib/views.py

```diff
@@ -49,6 +50,84 @@ def has_permission(self):
         return self.request.user.has_perms(perms, obj)
 
 
+class AutoPermissionRequiredMixin(PermissionRequiredMixin):
+    """
+    An extended variant of PermissionRequiredMixin which automatically determines
+    the permission to check based on the type of view it's used with.
+
+    It works by checking the current view for being an instance of a pre-defined
+    list of view types. On a match, the corresponding permission type (such as
+    "add" or "change") is converted into the full model-specific permission name
+    and checked. See the permission_type_map attribute for the default view type ->
+    permission type mappings.
+
+    When a view using this mixin has an attribute ``permission_type``, that type is
+    used directly and overwrites the permission_type_map for the particular view. A
+    permission type of ``None`` causes permission checking to be skipped. If the
```

Comment:
“permission type” should refer to the attribute here.

Reply:
`None` can either be set as `permission_type` directly or in the `permission_type_map`.